proposals/0095-AOA-multiplexing.md

```diff
@@ -1,72 +1,193 @@
+        ...
+    }
+}
+```
 
-So that, recommendation is choose __"1.1"__ as AOA parameter, to avoid user confusion.
+SdlRouterService selects either BluetoothTransportWriter or AoaTransportWriter according to the TransportType of the RegisteredApp instance:
+```java
+public class SdlRouterService extends Service {
+    ...
+    // TransportWriters -- for now BT and USB.
+	private BluetoothTransportWriter bluetoothTransportWriter;
+	private AoaTransportWriter aoaTransportWriter;
+    ...
+    // utilizes the TransportWriter in accordance with the TransportType for each registered app.
+}
+```
 
-Case Matrix is as follows:  
-In the case of New AOA parameter "SDL","Core",__"1.1"__:
-- Case: "old apps + new apps": "old apps" cannot be activated because of different accessory parameter. Only one new app can be router of AOA, other new apps can connect via router.
-- Case: "old apps" only: old apps __cannot__ connect.
-- Case: "new apps, no old apps": normal case, all apps can connect.
+## Potential downsides
 
-In the case of same AOA parameter "SDL","Core",__"1.0"__:
-- Case: "old apps + new apps": if end-user choose one of "old apps" in Android user confirmation dialog "which app should be used for this accessory" then, only one of old apps can connect. If end-user choose one of "new apps" then, all of "new apps" can connect, and old apps __cannot__ connect.
-- Case: "old apps" only: Only one of old apps can connect.
-- Case: "new apps, no old apps": normal case, all apps can connect.
+This feature introduces TransportType for TransportBroker and SdlRouterService. While this approach should have no obvious downsides, backward compatibility should be taken into account as much as we can.
+In particular, the following cases need to be confirmed:
+- Case #1: Older proxy's TransportBroker binds to new SdlRouterService. In this case, SdlRouterService assumes TransportType == MULTIPLEX for backward compatibility.
+- Case #2: Newer proxy's TransportBroker binds to older SdlRouterService. The older SdlRouterService won't support AOA multiplexing. In this case, the expected behavior is "don't bind to older SdlRouterService; instead, start and bind to newer (local) SdlRouterService". We can utilize existing version check and trusted router logic to make this case to work.
 
+This feature also increases the IPC transaction between TransportBroker and SdlRouterService. While Android system has Binder Transaction Limit, which is explained at [TransactionTooLargeException Android document](https://developer.android.com/reference/android/os/TransactionTooLargeException.html), we won’t run into TransactionTooLargeException cases in real scenario based on our test, unless underlying transport has a fatal error. The fatal error case would be, for example, the case where we cannot write Bluetooth socket and/or USB's ParcelFileDescriptor for some reason. Those fatal error cases can be discussed in outside of this proposal.
```

Comment:
Please update last sentence to "Those fatal error cases can be discussed outside of this proposal."